src/app/pages/register/register.component.ts

```diff
@@ -26,6 +26,7 @@ export class RegisterComponent {
       this.router.navigate(['/login']);
     })
     .catch((err) => { 
+      alert('Error Message:' + err);
       this.router.navigate(['/register']);
```

Comment:
You still need to get rid of this navigate. It doesn't do anything since they are already on the page. Also instead of using an alert, it may be best to have an error div that shows up somewhere on the webpage to alert the user that they could not register their account.